### PR TITLE
Added Elektor LoRa Node ( STM32F072 based )

### PR DESCRIPTION
**Pull request by lab-mathias-claussen**
This adds board support for Elektor LoRa Node with STM32F072 MCU

### CODE REVIEW

**Comment by fpistm** [comment]
Hi @lab-mathias-claussen 
thanks for this PR.
Please, could you add it under "Lora Boards" menu instead of create new item menu.
Could you provide a link the the board description (User manual, schematics,...)?
Thanks

**Comment by elektor-labs** [comment]
Hi, the board is moved, currently i struggle with the "board discription", i guess it's not meant in the baord.txt but somewhere here ?

**Comment by lab-mathias-claussen** [comment]
Hi, the board is moved, currently i struggle with the "board discription", i guess it's not meant in the baord.txt but somewhere here ?

**Comment by fpistm** [comment]
Thanks @lab-mathias-claussen 
anyway it seems you made a mistake when updating the boards.txt as it includes lot of changes not related to the new variant.

For the board description, this is mainly to review the variant (pin mapping, ...) and having a link to add in the README.md if this PR is merged.

**Comment by lab-mathias-claussen** [comment]
Hi,

the board.txt should be fixed now. I am sorry for the messed up board.txt.

File: variants/ELEKTOR_F072C8/PeripheralPins.c [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  {PA_1,  TIM15,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_TIM15, 1, 1)}, // TIM15_CH1N
+  {PA_2,  TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_TIM2, 3, 0)}, // TIM2_CH3
+  {PA_2,  TIM15,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF0_TIM15, 1, 0)}, // TIM15_CH1
+  {PA_3,  TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_TIM2, 4, 0)}, // TIM2_CH4
+  {PA_3,  TIM15,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF0_TIM15, 2, 0)}, // TIM15_CH2
+  {PA_4,  TIM14,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF4_TIM14, 1, 0)}, // TIM14_CH1
+  {PA_5,  TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_TIM2, 1, 0)}, // TIM2_CH1
+  {PA_6,  TIM3,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM3, 1, 0)}, // TIM3_CH1
+  {PA_6,  TIM16,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_TIM16, 1, 0)}, // TIM16_CH1
+  {PA_7,  TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_TIM1, 1, 1)}, // TIM1_CH1N
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/PeripheralPins.c]: To save space, I advise to comment duplicated pins definition. As only the first one found will be used, example to use the `TIM14` for `PA_7` you should do like this::
```
  // {PA_7,  TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_TIM1, 1, 1)}, // TIM1_CH1N
  // {PA_7,  TIM3,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM3, 2, 0)}, // TIM3_CH2
  {PA_7,  TIM14,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF4_TIM14, 1, 0)}, // TIM14_CH1
  // {PA_7,  TIM17,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_TIM17, 1, 0)}, // TIM17_CH1
```

File: variants/ELEKTOR_F072C8/PeripheralPins.c [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+//*** No QUADSPI ***
+
+//*** USB ***
+
+#ifdef HAL_PCD_MODULE_ENABLED
+WEAK const PinMap PinMap_USB[] = {
+  {PA_11, USB, STM_PIN_DATA(STM_MODE_INPUT, GPIO_NOPULL, GPIO_AF_NONE)}, // USB_DM
+  {PA_12, USB, STM_PIN_DATA(STM_MODE_INPUT, GPIO_NOPULL, GPIO_AF_NONE)}, // USB_DP
+  {PA_13, USB, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_USB)}, // USB_NOE
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/PeripheralPins.c]:
```suggestion
  // {PA_13, USB, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF2_USB)}, // USB_NOE
```

File: variants/ELEKTOR_F072C8/PeripheralPins.c (hunk 1 above)
Comment (lab-mathias-claussen) [reply variants/ELEKTOR_F072C8/PeripheralPins.c]: Duplicated PINs have been removed.

File: variants/ELEKTOR_F072C8/variant.h [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+#define  PF0    35
+#define  PF1    36
+
+
+// This must be a literal
+#define NUM_DIGITAL_PINS        37
+#define NUM_ANALOG_INPUTS       10
+#define NUM_ANALOG_FIRST        26
+
+// Below ADC, DAC and PWM definitions already done in the core
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/variant.h]: This is already define at 12.
```suggestion
```

File: variants/ELEKTOR_F072C8/variant.h [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+// Could be redefined here if needed
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/variant.h]:
```suggestion
```

File: variants/ELEKTOR_F072C8/variant.h [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+// ADC resolution is 12bits
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/variant.h]:
```suggestion
```

File: variants/ELEKTOR_F072C8/variant.h [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+#define ADC_RESOLUTION          12
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/variant.h]:
```suggestion
```

File: variants/ELEKTOR_F072C8/variant.h [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
```
Comment (fpistm) [inline variants/ELEKTOR_F072C8/variant.h]:
```suggestion
```

**Pull request merged**